pygments/lexers/typst.py

```diff
@@ -27,49 +34,74 @@ class TypstLexer(RegexLexer):
     url = 'https://typst.app'
     version_added = '2.18'
 
+    MATH_SHORTHANDS = (
+        '[|', '|]', '||', '*', ':=', '::=', '...', '\'', '-', '=:', '!=', '>>',
+        '>=', '>>>', '<<', '<=', '<<<', '->', '|->', '=>', '|=>', '==>',
+        '-->', '~~>', '~>', '>->', '->>', '<-', '<==', '<--', '<~~', '<~',
+        '<-<','<<-','<->','<=>','<==>','<-->', '>', '<', '~', ':',
+    )
+
     tokens = {
         'root': [
             include('markup'),
         ],
-        'common': [
-            (r'[ \t]+', Whitespace),
-            (r'((?!=[*_$`\-+0-9/<@\\#\[]|https?://).)+', Text),
+        # common cases going from math/markup into code mode
+        'into_code': [
+            (words(('#let', '#set', '#show'), suffix=r'\b'), Keyword.Declaration, 'inline_code'),
+            (words(('#import', '#include'), suffix=r'\b'), Keyword.Namespace, 'inline_code'),
+            (r'#\{', Punctuation, 'code'),
+            (r'#\(', Punctuation, 'code'),
+            (r'(#[a-zA-Z_][a-zA-Z0-9_-]*)(\[)', bygroups(Name.Function, Punctuation), 'markup'),
+            (r'(#[a-zA-Z_][a-zA-Z0-9_-]*)(\()', bygroups(Name.Function, Punctuation), 'code'),
+            (words(('#true', '#false', '#none', '#auto'), suffix=r'\b'), Keyword.Constant),
+            (r'#[a-zA-Z_][a-zA-Z0-9_]*', Name.Variable),
+            (r'#0x[0-9a-fA-F]+', Number.Hex),
+            (r'#0b[01]+', Number.Binary),
```

Comment:
Should be `Bin` according to https://github.com/pygments/pygments/blob/f64833d9d0443ff2443f8f8a4357c00196734d6c/pygments/token.py#L181

```suggestion
            (r'#0b[01]+', Number.Bin),
```